model_analyzer/result/model_config_measurement.py

```diff
@@ -0,0 +1,283 @@
+
+    def is_better_than(self, other):
+        """
+        Checks whether a measurement is better than
+        another
+
+        If True, this means this measurement is better
+        than the other.
+        
+        Parameters
+        ----------
+        other: ModelConfigMeasurement
+            set of (non_gpu) metrics to be compared against
+        """
+
+        # seems like this should be == -1 but we're using a min heap
+        return self._compare_measurements(other) == 1
+
+    def __eq__(self, other):
+        """
+        Check whether two sets of measurements are equivalent
+        
+        Parameters
+        ----------
+        other: ModelConfigMeasurement
+            set of (non_gpu) metrics to be compared against
+        """
+
+        return self._compare_measurements(other) == 0
+
+    def __lt__(self, other):
+        """
+        Checks whether a measurement is better than
+        another
+
+        This is used when sorting
+        
+        Parameters
+        ----------
+        other: ModelConfigMeasurement
+            set of (non_gpu) metrics to be compared against
+        """
+
+        return self.is_better_than(other)
+
+    def _compare_measurements(self, other):
+        """
+        Compares two ModelConfig measurements 
+        based on the weighted metric objectives
+
+        Parameters
+        ----------
+        other : ModelConfigMeasurement
+            set of (non_gpu) metrics to be compared against
+
+        Returns
+        -------
+        int
+            0
+                if the results are determined
+                to be the same within a threshold
+            1
+                if self > other (is better than)
+            -1
+                if self < other (is worse than)
+        """
+
+        weighted_score = self._calculate_weighted_score(other)
+
+        if weighted_score > COMPARISON_SCORE_THRESHOLD:
+            return 1
+        elif weighted_score < -COMPARISON_SCORE_THRESHOLD:
+            return -1
+        return 0
+
+    def _calculate_weighted_score(self, other):
+        """
+        Calculates the weighted score between two 
+        ModelConfig measurements based on the weighted
+        metric objectives
+        
+        Parameters
+        ----------
+        other : ModelConfigMeasurement
+            set of (non_gpu) metrics to be compared against
+            
+        Returns
+        -------
+        float
+            The weighted score. A positive value indicates 
+            this ModelConfig measurement is better than the other
+        """
+
+        weighted_score = 0.0
+        for objective, weight in self._metric_weights.items():
+            self_metric = self.get_metric(tag=objective)
+            other_metric = other.get_metric(tag=objective)
+
+            # Handle the case where objective GPU metric is queried on CPU only
+            # TODO: I left this code here, but I'm not 100% sure that it is
```

Comment:
We need to resolve this TODO

Reply:
I'd like to leave this for a future investigation as I don't think it's blocking any work. If you're okay with that I'll file a story for this.

Reply:
sure, file a story. It basically just handles "None" case in comparisons -- if one is valid and the other is None then the valid one is "better". If both None then they are considered equal. It looks like we can get None results for some GPU metrics when CPU_ONLY is true. Seems like you could easily add a test with a None metric to verify this.

Reply:
Makes sense. I've added testing to cover this and removed the TODO